vars/prebuiltQualityToolset.groovy

```diff
@@ -0,0 +1,97 @@
+/**
+ * Default behavior: delegate to {@link #basic}.
+ * <p>
+ *     Designed to work well with the Warnings Next Generation
+ *     plugin, as executed by e.g. {@link reportQuality#call}.
+ * </p>
+ * <p>
+ *     This method consumes <code>services</code>.
+ * </p>
+ *
+ * @param services a list of String codenames for specific tools
+ * @return a list of tools that make sense for this situation
+ */
+List call(Set<String> services) {
+	return basic(services)
+}
+
+/**
+ * This looks for one or so non-committal quality checks.
+ * <p>
+ *     Designed to work well with the Warnings Next Generation
+ *     plugin, as executed by e.g. {@link reportQuality#call}.
+ * </p>
+ * <p>
+ *     This method consumes <code>services</code>.
+ * </p>
+ *
+ * @param services a list of String codenames for specific tools
+ * @return a list of tools that make sense for this situation
+ * @see prebuiltQualityToolset#standardAndroidToolset
+ */
+List basic(Set<String> services) {
+	List toolset = []
+
+	// If this entry is present, scan for TODOs and FIXMEs.
+	if (services.remove("taskScanner")) {
+		toolset.add(
+				taskScanner(excludePattern: '**/build/**, **/node_modules/**, qualityReports/**', highTags: 'FIXME,suck', ignoreCase: true, includePattern: '**/*.swift, **/*.java, **/*.ts, **/*.kt, **/*.xml, **/*.m, **/*.h, **/*.c, **/*.yml, **/*.gradle', lowTags: 'deprecated', normalTags: 'TODO')
+		)
+	}
+	return toolset
+}
+
+/**
+ * Choose from a decent set of tools that we typically associate
+ * with Android projects.
+ * <p>
+ *     Designed to work well with the Warnings Next Generation
+ *     plugin, as executed by e.g. {@link reportQuality#call}.
+ * </p>
+ * <p>
+ *     This method consumes <code>services</code>.
+ * </p>
+ *
+ * @param services a list of String codenames for specific tools
+ * @return a list of tools that make sense for this situation
+ */
+List standardAndroidToolset(Set<String> services) {
+	// By default, we should always look for JavaDoc errors in the console output.
+	List toolset = [
+	        javaDoc()
+	]
+	// Minor sanity check in case the plugin API changes significantly
+	println "Toolset: " + toolset.getClass()
+
+	if (services.remove("cpd")) {
+		toolset.add(
+				cpd(highThreshold: 120, pattern: '**/cpd.xml, **/cpdCheck.xml', reportEncoding: 'UTF-8', skipSymbolicLinks: true)
```

Comment:
generally i think you over-complicated this so far. It means the developer having to configure more here. In reality these tools aren't performing any work just checking to see if a file exist. If it does it reports on it otherwise it does nothing so the level of separation is unnecessary. I would just make it do our suite of tools and it's up to the project and fazu/gradle and platform specific reporting suites to surface all the reports we can record.

Reply:
Hmm. I think you're right about overcomplicating, I'll try to do something about that. Could certainly have better function names.

The plugin _will_ add a line (or often a chart if we're not consolidating everything) to the status page for each enabled tool. So if we have 10 tools enabled it can be a little noisy. I don't expect the average dev to write a custom list of tools, or care about prebuiltQualityToolset - just to be able to copy-paste a prebuilt set and ignore it. I was also thinking maybe it'd make it easier to add new checks in future without automatically opting everyone in.

I'd make it 'do our suite of tools' if I could enforce something about what belongs in there on a per-platform basis. You wouldn't happen to know what quality checks iOS projects usually use?

Reply:
I figured out which checks to use! Resolving this conversation.